jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c

```diff
@@ -2458,8 +2459,11 @@ ecma_builtin_array_prototype_object_reduce (ecma_value_t this_arg, /**< this arg
     }
     else
     {
+      ecma_value_t accumulator = ECMA_VALUE_UNDEFINED;
+
       /* 6. */
-      uint32_t index = 0;
+      uint32_t index = start_from_left ? 0 : (len - 1);
```

Comment:
Ok my idea looks like the following. This variable is always initialized with 0.
`uint32_t index = 0;`